code/game/machinery/adv_med.dm

```diff
@@ -219,6 +223,7 @@
 		return
 	var/mob/living/carbon/human/H = occupant
 	var/list/occupant_data = list(
+		"name" = H.real_name,
```

Comment:
Revert/change this. Your identity is detected depending on your ID, as it is a machine. Have it be their name, not their real_name.